src/main/java/de/hysky/skyblocker/skyblock/dungeon/CroesusProfit.java

```diff
@@ -0,0 +1,272 @@
+            ItemStack stack = entry.getValue();
+            if (stack != null && stack.getNbt() != null) {
+                if (stack.equals(bestChest)) {
+                    highlights.add(ColorHighlight.green(entry.getKey()));
+                } else if (stack.equals(secondBestChest) && secondBestValue > dungeonKeyPriceData) {
+                    highlights.add(ColorHighlight.yellow(entry.getKey()));
+                }
+            }
+        }
+        return highlights;
+    }
+
+
+    private long valueChest(@NotNull ItemStack chest) {
+        long chestValue = 0;
+        int chestPrice = 0;
+        List<String> chestItems = new ArrayList<>();
+
+        Matcher matcher = CHEST_PATTERN.matcher(chest.getName().toString());
+        String chestType = (matcher.find() ? matcher.group(1) : "Error");     // we are doing it for sole reason of saving us headache of a spirit pet rarity
+
+        boolean processingContents = false;
+        for (Text line : chest.getTooltip(null, TooltipContext.BASIC)) {
+            String lineString = line.getString();
+            if (lineString.contains("Contents")) {
+                processingContents = true;
+                continue;
+            } else if (lineString.isEmpty()) {
+                processingContents = false;
+            } else if (lineString.contains("Coins") && !processingContents) {
+                chestPrice = Integer.parseInt(lineString.replaceAll(",", "").replaceAll("\\D", ""));
+            }
+
+            if (processingContents) {
+                if (lineString.contains("Essence")) {
+                    matcher = ESSENCE_PATTERN.matcher(lineString);
+                    if (matcher.matches()) {    // add to chest value result of multiplying price of essence on it's amount
+                        chestValue += getItemPrice(("ESSENCE_" + matcher.group("type")).toUpperCase()) * Integer.parseInt(matcher.group("amount"));
+                    }
+                } else {
+                    if (lineString.contains("Spirit")) {    // I hate pets in chests. Speaking of which, this is only part of code I'm really not sure of
+                        if (Objects.equals(chestType, "Diamond")) {
+                            chestValue += getItemPrice("Spirit Epic");      // Diamond chest containts epic spirit
+                        } else {
+                            chestValue += getItemPrice(lineString);
+                        }
+                    } else {
+                        chestItems.add(lineString);
+                    }
+                }
+            }
+        }
+        for (String item : chestItems){
+            chestValue += getItemPrice(item);
+        }
+        return chestValue-chestPrice;
+    }
+
+    private long getItemPrice(String itemDisplayName) {
+        JsonObject bazaarPrices = TooltipInfoType.BAZAAR.getData();
+        JsonObject lbinPrices = TooltipInfoType.LOWEST_BINS.getData();
```

Comment:
In the `TooltipInfoType` enum for the bazaar and lowest bins data add an or statement to the enabled predicate to make sure that the data is fetched in case a player has disabled all other features that depend on this data.

Reply:
I think I did it?